main.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/python3
+
+import json
+import boto3
+import os
+import gzip
+import shutil
+
+BUCKET_NAME = 'tf-creds-trail'
+DOWNLOAD_PATH = 'AWSLogs/203972369401/CloudTrail/eu-west-2/2022/12/13/'
+
+
+def download_items(bucket_name, download_path):
+    s3 = boto3.resource('s3')
+    s3_client = boto3.client('s3')
+    bucket = s3.Bucket(bucket_name)
+    my_bucket = s3_client.list_objects(
+        Bucket=bucket_name, Marker=download_path)['Contents']
+
+    for s3_object in my_bucket:
+        object_key = s3_object['Key']
+        file_name = os.path.basename(object_key)
+        # unzipped_file_name = os.path.splitext(object_key[0])
+
+        with open(file_name, '+wb') as f:
```

Comment:
Can the same context manager be used for both actions? Interesting to check if open can be set to write and read.